### PR TITLE
[ORCA-167] Stage files lazily

### PR DESCRIPTION
**Pull request by BrunoGrandePhD**
This PR takes advantage of the functionality implemented in https://github.com/Sage-Bionetworks-Workflows/py-dcqc/pull/25. First, it removes the dedicated `STAGE_TARGET` process. Second, for internal tests, it stages the file during the process that computes the test result. Some tests (_e.g._ `FileExtensionTest`) don't bother with staging the file since they only need the file name. Third, for external tests, the file is staged when the process is being prepared. For now, it cannot be staged in the process itself (_e.g._ in the libtiff container) because it might not have the required Python dependencies. Still, this dramatically reduces the amount of disk and network I/O. 

**Edit:** The CI tests are expected to fail until we update the `main` tag for the `py-dcqc` container. 

Depends on https://github.com/Sage-Bionetworks-Workflows/py-dcqc/pull/25

### CODE REVIEW

**Comment by BrunoGrandePhD** [comment]
Alright, the tests have passed!

File: conf/local.config [hunk 1]
```diff
@@ -0,0 +1,5 @@
+process {
+    withLabel:dcqc {
+        container   = 'dcqc'
```
Comment (thomasyu888) [inline conf/local.config:4]: Does this assume that you are building `py-dcqc` and naming the container `dcqc`?

File: modules/local/compute_test.nf [hunk 2]
```diff
@@ -14,6 +14,7 @@ process COMPUTE_TEST {
 
     script:
     """
+    export TMPDIR="./"
```
Comment (thomasyu888) [inline modules/local/compute_test.nf:17]: Nit: Without reading much more about it, can you let me know if the nextflow [scratch](https://www.nextflow.io/docs/latest/process.html#scratch) directive is relevant here?

**Review by thomasyu888 — APPROVED** [review]
🎸 Sweet!  Going to pre-approve as just some minor questions.

I ran this command and it passed:

```
nextflow run main.nf --input testdata/input_full.csv -profile docker
[a8/f6286d] process > SAGE_DCQC:DCQC:P... [100%] 1 of 1 ✔
[60/777fb1] process > SAGE_DCQC:DCQC:P... [100%] 11 of 11 ✔
[fd/3ee42b] process > SAGE_DCQC:DCQC:I... [100%] 22 of 22 ✔
[26/1b3460] process > SAGE_DCQC:DCQC:E... [100%] 17 of 17 ✔
[ba/dca3b6] process > SAGE_DCQC:DCQC:E... [100%] 17 of 17 ✔
[63/3ff464] process > SAGE_DCQC:DCQC:E... [100%] 17 of 17 ✔
[e2/fb70b8] process > SAGE_DCQC:DCQC:P... [100%] 11 of 11 ✔
[d0/31f75c] process > SAGE_DCQC:DCQC:P... [100%] 1 of 1 ✔
-[sage-bionetworks-workflows/nf-dcqc] Pipeline completed successfully-
Completed at: 03-Apr-2023 18:02:59
Duration    : 20m 12s
```

File: conf/local.config (hunk 1 above)
Comment (BrunoGrandePhD) [reply conf/local.config:4]: It's the default container image name when running `src/docker/build.sh` in the py-dcqc repo. I added a comment to clarify. 

https://github.com/Sage-Bionetworks-Workflows/py-dcqc/blob/main/src/docker/build.sh

File: modules/local/compute_test.nf (hunk 2 above)
Comment (BrunoGrandePhD) [reply modules/local/compute_test.nf:17]: When I wrote this, I looked into `scratch`, but it's not relevant. It has to do with changing where Nextflow is creating the work directory, whereas my goal is to change where Python is creating its temporary directories.

**Pull request merged**